### PR TITLE
check numbers in spatial parts

### PR DESCRIPTION
**Pull request by FuhuXia**
for https://github.com/GSA/data.gov/issues/3819

### CODE REVIEW

File: ckanext/geodatagov/logic.py [hunk 1]
```diff
@@ -470,7 +470,7 @@ def translate_spatial(old_spatial):
 
     # If we have 4 numbers separated by commas, transforme them as GeoJSON
     parts = old_spatial.strip().split(',')
-    if len(parts) == 4:
+    if len(parts) == 4 and all(x.replace('.', '', 1).isdigit() for x in parts):
```
Comment (nickumia-reisys) [inline ckanext/geodatagov/logic.py:473]: Why check `isdigit` instead of `float`?

**Review by nickumia-reisys — APPROVED** [review]
🏧

File: ckanext/geodatagov/logic.py (hunk 1 above)
Comment (jbrown-xentity) [reply ckanext/geodatagov/logic.py:473]: I see you pulled from https://stackoverflow.com/questions/354038/how-do-i-check-if-a-string-is-a-number-float. Personally I prefer the original post (see below), however I understand that the above is faster and contained in a single line. Maybe just put a comment noting why this is necessary, ie checking to make sure each part is a float (lat/long coordinate). That will help with the difficult reading of that line.
```
def is_number(s):
    try:
        float(s)
        return True
    except ValueError:
        return False
```

**Review by jbrown-xentity — APPROVED** [review]

**Comment by FuhuXia** [comment]
:) good feedbacks. ok i will change it to more official way of checking the number and not being afraid of being clucky by adding multiple lines of try/except codes.

**Pull request merged**